src/tls.c

```diff
@@ -114,16 +125,16 @@ static void get_server_name(
                             uint16_t extension_len = (tls[index] << 8 & 0xFF00) + (tls[index + 1] & 0x00FF);
                             index += 2;
                             // skip to the next extension, if there is one
-                            index += extensions_len;
+                            index += extension_len;
```

Comment:
This was really the bug, I was using `extensions_len` instead of (singular) `extension_len` 🤦